js/compiler.ts

```diff
@@ -84,7 +84,8 @@ export class ModuleMetaData implements ts.IScriptSnapshot {
   constructor(
     public readonly fileName: ModuleFileName,
     public readonly sourceCode = "",
-    public outputCode = ""
+    public outputCode = "",
+    public moduleName = ""
```

Comment:
I would change the order... `moduleName` shouldn't be optional and should be either first or second, I think it should also be `public readonly`.

Also, would `moduleId` be better here.  That is more commonly what it is referred to as.  It should also be of type `ModuleId` which should be a type alias for `string`.  So it would mean the following:

* `fileName` - An absolute resource identifier, which is often the local cache file name.
* `moduleId` - The absolute module ID, which could be a URL or a local filename.
* `moduleSpecifier` - A potentially relative module name, which is resolved in relationship to...
* `containingFile` - The containing module of which the module specifier is resolved against.